ksql-engine/src/main/java/io/confluent/ksql/function/udaf/topkdistinct/TopkDistinctAggFunctionFactory.java

```diff
@@ -39,13 +39,13 @@ public KsqlAggregateFunction getProperAggregateFunction(List<Schema> argTypeList
     Schema argSchema = argTypeList.get(0);
     switch (argSchema.type()) {
       case INT32:
-        return new TopkDistinctKudaf<Integer>(-1, 0, Integer.class);
+        return new TopkDistinctKudaf<>(-1, 0, Schema.INT32_SCHEMA, Integer.class);
       case INT64:
-        return new TopkDistinctKudaf<Long>(-1, 0, Long.class);
+        return new TopkDistinctKudaf<>(-1, 0, Schema.INT64_SCHEMA, Long.class);
       case FLOAT64:
-        return new TopkDistinctKudaf<Double>(-1, 0, Double.class);
+        return new TopkDistinctKudaf<>(-1, 0, Schema.FLOAT64_SCHEMA, Double.class);
       case STRING:
-        return new TopkDistinctKudaf<String>(-1, 0, String.class);
+        return new TopkDistinctKudaf<>(-1, 0, Schema.STRING_SCHEMA, String.class);
```

Comment:
The problem with the Schema stuff is that there is loads of nut-and-bolts style code scattered everywhere. We talked about stripping it out to use enums to handle these types of statements and then inject a lambda to create the function. It out of scope of this PR but everytime I see the schema class it IRKs me....

Reply:
@bluemonk3y - yes, tell me about it. However as this is for 4.1 we don't want to make big changes